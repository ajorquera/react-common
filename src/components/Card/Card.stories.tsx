import { MultipleComponentsTemplate } from '../../StoryTemplates';
import Card, { type Props } from './Card';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Card> = {
    title: 'Components/Card',
    component: MultipleComponentsTemplate(Card),
    decorators: [],
    parameters: {}
};

export default meta;

type Story = StoryObj<{ items: Props[] }>;

export const Default: Story = {
    args: {
        items: [
            {
                children: 'Card'
            },
            {
                children: '2nd Card'
            }
        ]
    }
};
