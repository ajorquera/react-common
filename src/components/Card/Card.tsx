import styled from 'styled-components'

import { Box, type BoxProps } from '../Box'
import { shadow, type ShadowProps } from 'styled-system'

export interface Props extends ShadowProps, BoxProps { }

const Card = styled(Box)<Props>(shadow)

Card.defaultProps = {
  boxShadow: '0 0.5em 1em -0.125em rgb(10 10 10 / 10%), 0 0px 0 1px rgb(10 10 10 / 2%)',
  bg: 'white',
  p: 3
}

export default Card
