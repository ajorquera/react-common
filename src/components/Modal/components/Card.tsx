import { Card } from '../../Card';
import styled from 'styled-components';

const CustomCard = styled(Card)`
    position: fixed;
    left: 50%;
    top: 50%;
    z-index: 101;
    transform: translate(-50%, -50%);
    overflow-x: auto;
`;

export default CustomCard;
