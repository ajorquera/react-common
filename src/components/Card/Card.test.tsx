import React from 'react'
import Card from './Card'

import { render } from '@testing-library/react'

test('should render component', () => {
  const { container } = render(<Card>My card</Card>)

  expect(container).toMatchSnapshot()
})
