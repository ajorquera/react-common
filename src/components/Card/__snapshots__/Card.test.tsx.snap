// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<div>
  <div
    bg="white"
    boxshadow="0 0.5em 1em -0.125em rgb(10 10 10 / 10%), 0 0px 0 1px rgb(10 10 10 / 2%)"
    class="sc-bdnyFh sc-dlniIP iaFBTH ffhxQx"
    p="3"
  >
    My card
  </div>
</div>
`;
